pdns/opensslsigners.cc

```diff
@@ -346,13 +346,13 @@ int OpenSSLRSADNSCryptoKeyEngine::hashSizeToKind(const size_t hashSize)
 
 std::string OpenSSLRSADNSCryptoKeyEngine::sign(const std::string& msg) const
 {
-  string hash = this->hash(msg);
-  int hashKind = hashSizeToKind(hash.size());
+  string l_hash = this->hash(msg);
+  int hashKind = hashSizeToKind(l_hash.size());
   std::string signature;
   signature.resize(RSA_size(d_key.get()));
   unsigned int signatureLen = 0;
 
-  int res = RSA_sign(hashKind, reinterpret_cast<unsigned char*>(&hash.at(0)), hash.length(), reinterpret_cast<unsigned char*>(&signature.at(0)), &signatureLen, d_key.get());
+  int res = RSA_sign(hashKind, (unsigned char*)l_hash.c_str(), l_hash.length(), (unsigned char*)signature.c_str(), &signatureLen, d_key.get());
```

Comment:
I'm not a big fan of the C++-style cast to C-style cast change, here..

Reply:
I'm also not sure why this commit also moves from `&.data(0)` to `c_str()`?

Reply:
because it seemed to be in minority. i think it would be sensible to do either one instead of doing both ways. i'm happy to rectify the others to use data(0) instead.

Reply:
also RSA_sign is C style API.

Reply:
ping?

Reply:
OK, I don't care much about `data()` vs `c_str()`,although I find the latter a bit misleading for binary strings, but I was surprised to see this change in a PR/commit about shadowed variables.
Regarding the casts, my humble point of view is that using a C-style cast is a regression in code readability, and I'm also not sure I see the relation with `RSA_sign()`?

Reply:
Ok, so unrelated change. I'll revert it.